File: examples/sort/evaluator.py
import random

from turingarena import *

all_passed = True
for _ in range(10):
    n = 20
    a = random.choices(range(10 ** 4, 10 ** 5), k=n)

    try:
        with run_algorithm(submission.source) as process:
            process.procedures.sort(n, a)
            b = [process.functions.get_element(i) for i in range(n)]
    except AlgorithmError as e:
        print(e)
        all_passed = False
    if b == sorted(a):
        print("correct!")
    else:
        print("WRONG!")
        all_passed = False

evaluation.data(dict(goals=dict(correct=all_passed)))

File: examples/sort/extra_evaluators/evaluator.cpp
#include "turingarena.h"

#include <iostream>
#include <array>

int main() {
    std::array<int, 10> A{4, 5, 1, 2, 8, 9, 0, 3, 7, 6};
    turingarena::Algorithm algorithm{turingarena::get_submission_parameter("source")};
    std::cerr << "Algorithm started\n";
    algorithm.call_procedure("sort", A.size(), A);
    for (int i = 0; i < A.size(); i++) {
        int e = algorithm.call_function("get_element", i);
        if (i != e) {
            std::cout << "WRONG" << " " << i << " " << e << '\n';
            return 0;
        }
    }
    std::cout << "CORRECT\n";
}
